+bids/File2.m

```diff
@@ -155,12 +165,25 @@
         end
       end
 
-      if ~isempty(obj.suffix)
+      obj.check_required_entities();
+
+      if isempty(obj.suffix)
+        obj.bidsFile_error('emptySuffix');
+      else
         filename = [filename obj.suffix];
       end
 
-      obj.filename = [filename obj.extension];
+      if isempty(obj.extension)
+        obj.bidsFile_error('emptyExtension');
```

Comment:
See my comment on the tests to see if when we want to throw a warning or error when there is no extension.